pkg/testsuite/mock/node.go

```diff
@@ -148,13 +157,21 @@ func (n *Node) Initialize(failOnBlockFiltered bool, opts ...options.Option[proto
 }
 
 func (n *Node) hookEvents() {
-	events := n.Protocol.Events
-
-	events.ChainManager.ForkDetected.Hook(func(fork *chainmanager.Fork) { n.forkDetectedCount.Add(1) })
+	n.Protocol.Chains.HeaviestAttestedCandidate.OnUpdate(func(_ *protocol.Chain, heaviestAttestedCandidate *protocol.Chain) {
+		if heaviestAttestedCandidate != nil {
+			n.forkDetectedCount.Add(1)
 
-	events.CandidateEngineActivated.Hook(func(e *engine.Engine) { n.candidateEngineActivatedCount.Add(1) })
+			heaviestAttestedCandidate.Engine.OnUpdate(func(prevEngine *engine.Engine, newEngine *engine.Engine) {
+				n.candidateEngineActivatedCount.Add(1)
+			})
```

Comment:
Just for my understanding: `heaviestAttestedCandidate.Engine` is only set when we start syncing this candidate chain? Before it is nil?

Reply:
Correct.